hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -895,7 +895,9 @@ public boolean delete(final Path f, final boolean recursive) throws IOException
         return false;
       }
 
-      return deleteRoot();
+      if (getAbfsStore().getPrefixMode() == PrefixMode.DFS) {
```

Comment:
Can you please add a comment for this special handling ?

Reply:
added
```
/*
       * For DFS root directory deletion, direct DELETE Path API on root doesn't
       * work, the server returns: "The request URI is invalid.", 400, DELETE.
       * Hence, there is a special handling for root deletion on DFS endpoint
       */
```